internal/controller/component_build_controller_service_account.go

```diff
@@ -0,0 +1,523 @@
+// that should be used for build pipelines of the given Component.
+func getBuildPipelineServiceAccountName(component *appstudiov1alpha1.Component) string {
+	return "build-pipeline-" + component.GetName()
+}
+
+// EnsureBuildPipelineServiceAccount checks if dedicated Service Account for Component build pipelines and
+// corresponding Role Binding to appstudio-pipeline-runner Cluster Role exists.
+// If a resource missing, it will be created.
+func (r *ComponentBuildReconciler) EnsureBuildPipelineServiceAccount(ctx context.Context, component *appstudiov1alpha1.Component) error {
+	log := ctrllog.FromContext(ctx).WithName("buildSA-provision")
+	ctx = ctrllog.IntoContext(ctx, log)
+
+	// Verify build pipeline Service Account
+	buildPipelineServiceAccountName := getBuildPipelineServiceAccountName(component)
+	buildPipelinesServiceAccount := &corev1.ServiceAccount{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: buildPipelineServiceAccountName, Namespace: component.Namespace}, buildPipelinesServiceAccount); err != nil {
+		if !errors.IsNotFound(err) {
+			log.Error(err, fmt.Sprintf("failed to read build pipeline Service Account %s in namespace %s", buildPipelineServiceAccountName, component.Namespace), l.Action, l.ActionView)
+			return err
+		}
+
+		// Service Account for Component build pipelines doesn't exist.
+		// Create Service Account for the build pipeline.
+		buildPipelineSA := &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      buildPipelineServiceAccountName,
+				Namespace: component.Namespace,
+			},
+		}
+		if err := controllerutil.SetOwnerReference(component, buildPipelineSA, r.Scheme); err != nil {
+			log.Error(err, "failed to add owner reference to build pipeline Service Account")
+			return err
+		}
+		if err := r.linkCommonSecretsToBuildPipelineServiceAccount(ctx, buildPipelineSA); err != nil {
+			return err
+		}
+		if err := r.Client.Create(ctx, buildPipelineSA); err != nil {
+			log.Error(err, fmt.Sprintf("failed to create Service Account %s in namespace %s", buildPipelineServiceAccountName, component.Namespace), l.Action, l.ActionAdd)
+			return err
+		}
+	}
+
+	if err := r.ensureBuildPipelineServiceAccountBinding(ctx, component); err != nil {
+		log.Error(err, "failed to ensure role binding for build pipleine Service Account")
+		return err
+	}
+
+	return nil
+}
+
+// linkCommonSecretsToBuildPipelineServiceAccount searches for common build Secrets in Component namespace
+// marked with corresponding label and adds them into secrets section of the given Service Account.
+func (r *ComponentBuildReconciler) linkCommonSecretsToBuildPipelineServiceAccount(ctx context.Context, serviceAccount *corev1.ServiceAccount) error {
+	// Get all secrets with common for all Component label
+	commonSecretsList := &corev1.SecretList{}
+
+	commonBuildSecretRequirement, err := labels.NewRequirement(commonBuildSecretLabelName, selection.Equals, []string{"true"})
```

Comment:
what is setting that annotation?

Reply:
According to the design doc, UI should set it along with linking the secret to all existing Components if user marks a secret as common for all existing and new Components.